http/error.go

```diff
@@ -81,6 +81,9 @@ func (resp *ErrorResponse) MarshalXML(e *xml.Encoder, _ xml.StartElement) error
 // error. This method is used by the generated server code when the error is not
 // described explicitly in the design.
 func (resp *ErrorResponse) StatusCode() int {
+	if resp.Name == ErrUnsupportedMediaTypeName {
+		return http.StatusUnsupportedMediaType
+	}
```

Comment:
It seems that 400 is returned instead of 415 because the error name is overwritten by `DecodePayloadError()`.

https://github.com/goadesign/goa/blob/feature/unsupported_media_type/http/codegen/templates/request_decoder.go.tpl#L25

 If the error is `ServiceError`, it must be returned as is.

```go
var gerr *goa.ServiceError
if errors.As(err, &gerr) {
	return nil, err
}
```

Reply:
Great catch! I have modified the generated code to follow the suggestion above.